src/app/dataset-block/metadata-block/components/event-details/components/set-polling-source-event/set-polling-source-event.component.ts

```diff
@@ -0,0 +1,105 @@
+import { EventRow } from "./../../factory.events";
+import { DataHelpers } from "src/app/common/data.helpers";
+import {
+    FetchStepFilesGlob,
+    FetchStepUrl,
+    MergeStrategyLedger,
+    ReadStepCsv,
+    SetPollingSource,
+} from "./../../../../../../api/kamu.graphql.interface";
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ComponentRef,
+    Input,
+    OnInit,
+    QueryList,
+    ViewChildren,
+    ViewContainerRef,
+} from "@angular/core";
+import { DatasetInfo } from "src/app/interface/navigation.interface";
+import { sqlEditorOptionsForEvents } from "../../config-editor.events";
+import { LogoInfo } from "../../supported.events";
+import { FACTORIES_BY_EVENT_TYPE } from "../../factory.events";
+import { BasePropertyComponent } from "../common/base-property/base-property.component";
+
+@Component({
+    selector: "app-set-polling-source-event",
+    templateUrl: "./set-polling-source-event.component.html",
+    styleUrls: ["./set-polling-source-event.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SetPollingSourceEventComponent implements OnInit, AfterViewInit {
+    @Input() public event: SetPollingSource;
+    @Input() public datasetInfo: DatasetInfo;
+    @ViewChildren("readStepContainer", { read: ViewContainerRef })
+    readStepContainer: QueryList<ViewContainerRef>;
+    public isYamlView = false;
+    public sqlEditorOptions = sqlEditorOptionsForEvents;
+    public readStepCsvData: EventRow[];
+    private componentRef: ComponentRef<BasePropertyComponent>;
```

Comment:
Logicaly, it's rather a local temporary variable within `ngAfterViewInit`, than a long-lived property